plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/jobs/MutexRule.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.util.jobs;
+
+import org.eclipse.core.runtime.jobs.ISchedulingRule;
+
+/**
+ * Scheduling rule to prevent two jobs from running concurrently. Set this to jobs that need to run
+ * serially.
+ */
+public class MutexRule implements ISchedulingRule {
```

Comment:
One suggestion might be to provide a descriptive string, output in its `toString()`, to see what this mutex rule is about.  It might help us one day.

Reply:
There seems nothing much to say. There is no field, no internal state. The built-in `toString()` will print the full qualified class name, and that's basically all the info?

Reply:
OK, maybe I could say

`return MutexRule.class + ": mutex rule for running jobs mutually exclusively`

Reply:
I was thinking along the lines of `new MutexRule("ManagedCloudSDK")` and print that string in the toString().

Reply:
Good idea.